pandas/tests/test_index.py

```diff
@@ -938,6 +938,10 @@ def test_difference(self):
         self.assertEqual(len(result), 0)
         self.assertEqual(result.name, first.name)
 
+        # empty difference for subtypes
+        result = self.periodIndex.difference(self.periodIndex)
+        self.assertIsInstance(result, pd.PeriodIndex)
+
     def test_symmetric_diff(self):
```

Comment:
I've moved it there; that's a much better place, thanks.

As a result, I kept the type check, rather than a direct result vs expected comparison. Which is better? If we do a comparison, how would we create the expected? Different types are going to need different constructors (e.g. PeriodIndex would need a freq), unless we just did `._shallow_copy`, which seems like just testing that the code is running rather than working?